### ISSUE TITLE
Missing street numbers in combination with common street names can give confusing results

### ISSUE THREAD

**Issue by agunnarsson** [twain47/Nominatim#422]
Some searches for with the combination street name, street number and city don't contain a result with the expected street. The reason is that a street number match is "blocking" other streets in the result.

An example is "Schlesierstraße 11 München", where this address is found http://www.openstreetmap.org/node/1658288193 while 11 of this street is the expected result http://www.openstreetmap.org/way/141160515

Possible solutions can be:
1. Create interpolations of all existing street numbers. Numbers outside known ranges would still be missing though. Has there been any efforts to include street numbers on buildings and nodes surrounding a street in interpolation?
2. Include streets without a matching street number in the result. This would possibly hide the exact matches if those are not sorted to top of the result.

**Comment by lonvia** [comment]
To be precise, the issue with your example is not the street number itself.  Nominatim is capable of ignoring it, as you'll see if you check out  'more results' for that particular query.

Solution 2 might actually lead to some improvement if the results with the house number are given a proper importance boost, so that they are more likely to show up on top. Although looking at the results the difference in importance between the streets in München and in the Landkreis of München is fairly small. We probably need a better algorithm to weigh address importance first.

**Comment by agunnarsson** [comment]
No, exactly the problem is that the current implementation give you either any existing street numbers or streets if no number is found.

I will try to implement solution 2. I'm not familiar with the sorting though. I guess it's a combination of importance and how well something is matching, and that a boost might be needed in the latter part.

What about improved interpolation, where address tags on features surrounding a street are included? I suspect it will be hard to find an efficient solution which supports continuous updates. Any ideas?
